src/main/java/com/github/maleksandrowicz93/cqrsdemo/student/DeleteStudentCommandHandler.java

```diff
@@ -0,0 +1,18 @@
+package com.github.maleksandrowicz93.cqrsdemo.student;
+
+import com.github.maleksandrowicz93.cqrsdemo.student.exception.StudentNotFoundException;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+class DeleteStudentCommandHandler {
+
+    private final StudentRepository studentRepository;
+
+    boolean handle(long studentId) throws StudentNotFoundException {
```

Comment:
dziwne API - zwracany boolean (zawsze true) lub rzucany wyjątek, jeśli nie znaleziono

Reply:
Też miałem mieszane uczucia pisząc to (kiedyś słyszałem, że warto się kierować emotions driven development :P).
Skąd taka decyzja? Wydaje mi się, że miałem 2 opcje: 
1. Zwrócić void i rzucać exc
2. Zwracać true/false
Osobiście nie lubię void. Wolę jeśli metoda coś zwraca. Void pachnie też potencjalnymi skutkami ubocznymi.
Z tego tytułu skierowałem się w stronę true/false.
ALE!
Niektóre endpointy zwracają wartość lub ErrorResponse.
A inne w wyniku tych samych warunków gdzieś tam wewnątrz zwracałyby false.
Więc pewne funkcjonalności w ramach tej samej sytuacji rzucają wyjątkiem, a inne zwracają false.
Bardzo mi się coś takiego nie podoba, więc chciałem zwrócić cokolwiek, ale nie zaburzać konsystentności, tego, czego można się spodziewać po API.

Niemniej jednak, ten mój "potworek" wygląda mega dziwnie, chyba jeszcze dziwniej od tego, przed czym chciałem uciec.
Osobiście chyba się skłonię ku void + wyjątek i jakoś tego aoida postaram się przeżyć :)

Reply:
Propozycja - napisz od zera `AddStudentCommandHandler` zaczynając od testów, w międzyczasie skomponuj API pod testy, ale nie implementuj metody. Skup się na tym, żeby z perspektywy testu było Ci jak najwygodniej i jak najbardziej naturalnie. Następnie zaimplementuj logikę, żeby testy przeszły. Zapomnij o tym, co napisałeś dotychczas w związku z tym handlerem, skup się na teście i wymaganiach. Na wszelki wypadek napiszę jeszcze coś, co z perspektywy konsumenta API (testu) powinno być naturalne: nie rzucaj żadnych wyjątków :p

Reply:
Na razie jest void z ewentualnymi exceptions. Refactor API będzie w kolejnej iteracji, zgodnie z komentarzem: https://github.com/maleksandrowicz93/cqrs-demo-apprvd/pull/1/files#r1104401292